File: [23.12]Design-Pattern/Java/src/creational/enumFactoryMethod/Main.java
package creational.enumFactoryMethod;

public class Main {
    public static void main(String[] args) {
        System.out.println(EnumFoodFactory.DRINK.getName());
        EnumFoodFactory.DRINK.createFood();

        System.out.println("---------------------------------------");

        System.out.println(EnumFoodFactory.HAMBURGER.getName());
        EnumFoodFactory.HAMBURGER.createFood();


    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/creational/enumFactoryMethod/Main.kt
package creational.enumFactoryMethod

fun main() {
    println(FoodFactory.DRINK.name)
    println(FoodFactory.DRINK.foodName)
    FoodFactory.DRINK.createFood()
    println("-----------------------------")
    println(FoodFactory.HAMBURGER.name)
    println(FoodFactory.HAMBURGER.foodName)
    FoodFactory.HAMBURGER.createFood()
}